src/test/java/org/openrewrite/java/micronaut/AddMicronautRetryDependencyIfNeededTest.java

```diff
@@ -20,107 +20,94 @@
-            }
-            
-            repositories {
-                mavenCentral()
-            }
-            
-            dependencies {
-                implementation "io.micronaut:micronaut-retry"
-            }
-        """;
-
-    @Language("xml")
-    private final String pomInitial = """
-            <project>
-                <groupId>com.mycompany.app</groupId>
-                <artifactId>my-app</artifactId>
-                <version>1</version>
-                <parent>
-                    <groupId>io.micronaut.platform</groupId>
-                    <artifactId>micronaut-parent</artifactId>
-                    <version>4.0.0</version>
-                </parent>
-            </project>
-        """;
-
-    @Language("xml")
-    private final String pomExpected = """
-            <project>
-                <groupId>com.mycompany.app</groupId>
-                <artifactId>my-app</artifactId>
-                <version>1</version>
-                <parent>
-                    <groupId>io.micronaut.platform</groupId>
-                    <artifactId>micronaut-parent</artifactId>
-                    <version>4.0.0</version>
-                </parent>
-                <dependencies>
-                    <dependency>
-                        <groupId>io.micronaut</groupId>
-                        <artifactId>micronaut-retry</artifactId>
-                    </dependency>
-                </dependencies>
-            </project>
-        """;
+          import jakarta.inject.Singleton;
+          import io.micronaut.retry.annotation.Retryable;
+          
+          @Singleton
+          public class PersonService {
+          
+              @Retryable
+              public void callFlakyRemote() {
+                  
+              }
+          }
+      """;
 
     @Test
     void updateGradleDependencies() {
-        rewriteRun(spec -> spec.beforeRecipe(withToolingApi()), mavenProject("project", srcMainJava(java(retryableService)), gradleProperties, buildGradle(buildGradleInitial, buildGradleExpected)));
+        rewriteRun(spec -> spec.beforeRecipe(withToolingApi()), mavenProject("project", srcMainJava(java(retryableService)), getGradleProperties(),
+          //language=groovy
+          buildGradle(String.format("""
+                plugins {
+                    id("io.micronaut.application") version "%s"
+                }
+                
+                repositories {
+                    mavenCentral()
+                }
+            """, latestApplicationPluginVersion), String.format("""
+                plugins {
+                    id("io.micronaut.application") version "%s"
+                }
+                
+                repositories {
+                    mavenCentral()
+                }
+                
+                dependencies {
+                    implementation "io.micronaut:micronaut-retry"
+                }
+            """, latestApplicationPluginVersion))));
```

Comment:
We use Java 17 for our tests, so you could also use `"".formatted()`, but before you go make that change by hand we also have a recipe to do that for you: https://docs.openrewrite.org/recipes/java/migrate/lang/stringformatted
We can also choose to run that after a merge through the platform, as that should be even easier.

```suggestion
          buildGradle(String.format("""
                plugins {
                    id("io.micronaut.application") version "%s"
                }
                
                repositories {
                    mavenCentral()
                }
            """, latestApplicationPluginVersion), String.format("""
                plugins {
                    id("io.micronaut.application") version "%s"
                }
                
                repositories {
                    mavenCentral()
                }
                
                dependencies {
                    implementation "io.micronaut:micronaut-retry"
                }
            """, latestApplicationPluginVersion))));
```

Reply:
Ahh, of course, forgot about .formatted() being a thing now. I think there are still changes to be made here, so I'll try running that recipe before my next commit(s).